txmanager_integration_test.go

```diff
@@ -347,4 +347,44 @@ func TestTxManager_WithTransaction(t *testing.T) {
 		require.Nil(t, bookResult.ID)
 	})
 
+	t.Run("ReturnError_WhenContextCancel", func(t *testing.T) {
```

Comment:
@bintangsibarani IMO it's better to name the test to: "ContextCancelled_ThenRollback" or "Rollback_WhenContextCancelled". 

Also need to add assertion that the data is not written / rolled back.